### DISCUSSION TITLE
Toggle Smartshift from API

### DISCUSSION THREAD

**Discussion by rob0101** [amberelectric/public-api#228] [Ideas]
This has been asked for before but a long time ago.

Any chance we can get the API to offer an on/off toggle for smartshift ?

Most of the time I like to have smartshift off because my goal is battery management, not maximising income. There are times when I know it can do a better job than via other means.

As an example, I am far more conservative on when to sell from my battery than Smartshift's battery booster mode, so if I feel the price is good to sell, Smartshift will already be keen to do so too.

Currently if I want to sell I need to flip my Powerwall to TBC mode and hope that Netzero has already sent through current pricing. If the prices are old, TBC might do the exact opposite of what I want.  Smartshift has the advantage of knowing the prices in real-time and having direct control of a Powerwall, changing it from one mode to another in a couple of seconds.

The Facebook groups are full of examples of people manually toggling Smartshift on and off at certain times of the day to work around the times when it doesn't behave the way they want.   Often they need to do this in the middle of the night.

An on/off toggle sounds pretty straightforward to offer, and I can't think of any negative scenarios other than rapid toggling which could be prevented with a minimum time between opposite requests.

Is this feature likely to happen or should we forget about it ?

**Comment by madpilot** [comment]
It will happen, but it's waiting on a more nuanced mechanism for enabling different parts of the automation as well as overriding them.

I'm not going to implement it until that other work is done.

**Reply by chiangy77** [reply]
I'm curious what the rough timeline for this more nuanced mechanism and subsequent work would be? Also, would you guys consider API automation a high priority?

**Reply by madpilot** [reply]
> I'm curious what the rough timeline for this more nuanced mechanism and subsequent work would be?

We tend to not put deadlines or timelines on these sorts of things, unless they are the top priority for the quarter. But we have work that is in flight, so there is progress always happening.

> Also, would you guys consider API automation a high priority?

There are other things that are higher priority at the moment.

**Comment by purcell-lab** [comment]
#165
